opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/__init__.py

```diff
@@ -369,16 +370,20 @@ def __init__(
         self._shutdown_once = Once()
         self._shutdown = False
 
-    def force_flush(self) -> bool:
-
-        # FIXME implement a timeout
+    def force_flush(self, timeout_millis: float = 10_000) -> bool:
+        deadline_ns = _time_ns() + timeout_millis * 10**6
 
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader.collect()
+            current_ts = _time_ns()
+            if current_ts >= deadline_ns:
+                raise Exception("Timed out while flushing metric readers")
```

Comment:
Curious as to why we are raising exceptions if the flush/shutdown fails?

Reply:
Ya I think so, but I don't want to do it in this PR. OPened https://github.com/open-telemetry/opentelemetry-python/issues/2665